### PR TITLE
add vesting start time for class

### PR DESCRIPTION
**Pull request by decadanse**

### CODE REVIEW

File: test/seed.js [hunk 1]
```diff
@@ -693,15 +743,39 @@ describe("Contract: Seed", async () => {
...
               fee
           );
+          await setup.data.seed
+              .connect(admin)
+              .addClass(hardCap, CLASS_PERSONAL_FUNDING_LIMIT, price, 10000000, 1700000000);
 
           await setup.data.seed
               .connect(admin)
-              .addClass(1e14, 1e12, 1e12, 10000000);
+              .addClass(1e14, 1e12, 1e12, 10000000, 1700000000);
```
Comment (BritikovKI) [inline test/seed.js:752]: Can you pls transform all numeric values into constants in the beginning of the file?
It was my bad, I totally forgot to do it when I was writing tests

File: contracts/seed/Seed.sol [hunk 2]
```diff
@@ -20,6 +20,7 @@
 pragma solidity 0.8.9;
 
 import "openzeppelin-contracts-sol8/token/ERC20/IERC20.sol";
+import "hardhat/console.sol";
```
[the remark this answers is not in the document]
Comment (decadanse) [reply contracts/seed/Seed.sol]: Done

File: contracts/seed/Seed.sol [hunk 3]
```diff
@@ -177,6 +181,15 @@ contract Seed {
         // (seedAmountRequired*fee) / (100*FEE_PRECISION) = (seedAmountRequired*fee) / PRECISION
         //  where FEE_PRECISION = 10**16
         feeAmountRequired = (seedAmountRequired * fee) / PRECISION;
+        classes.push( ContributorClass(
+                hardCap,
+                hardCap,
+                _price,
+                _vestingDuration,
+                _endTime,
```
[the remark this answers is not in the document]
Comment (decadanse) [reply contracts/seed/Seed.sol]: changed to vestingStartTime, which is changed to vestingStartTime = endTime + 1;

File: test/seed.js (hunk 1 above)
Comment (decadanse) [reply test/seed.js]: Created 
  const CLASS_VESTING_DURATION =  10000000;
  const CLASS_VESTING_START_TIME = 1700000000;

Comment (BritikovKI) [reply test/seed.js]: Can you do the same for 1e14 and 1e12  pls?

File: test/seed.js [hunk 4]
```diff
@@ -480,7 +485,7 @@ describe("Contract: Seed", async () => {
 
             await alternativeSetup.seed
                 .connect(admin)
-                .addClass(1e14, 1e12, 1e12, 10000000);
+                .addClass(1e14, 1e12, 1e12, CLASS_VESTING_DURATION, CLASS_VESTING_START_TIME);
```
Comment (BritikovKI) [inline test/seed.js:488]: I think you forgot to change 1e14, 1e12 to constants, or just didn't push the commit...

Comment (decadanse) [reply test/seed.js]: they were pushed into 19 branch, as for those time this and 19 branches weren't connected... 
added only them here also

File: contracts/seed/Seed.sol [hunk 5]
```diff
@@ -640,13 +668,15 @@ contract Seed {
...
-    ) public view returns(uint256, uint256, uint256, uint256, uint256, uint256, uint256) {
-        return (classes[_id].classCap,
-                classes[_id].individualCap,
-                classes[_id].price,
-                classes[_id].vestingDuration,
-                classes[_id].fundingCollected,
-                classes[_id].seedAmountRequired,
-                classes[_id].feeAmountRequired);
+    ) public view returns(uint256, uint256, uint256, uint256, uint256, uint256, uint256, uint256) {
+        uint8 id = _id;
```
Comment (BritikovKI) [inline contracts/seed/Seed.sol:672]:
```suggestion

```

**Review by BritikovKI — APPROVED** [review]
Ledgit

**Review by arseneeth — APPROVED** [review]

**Pull request merged**